src/mapblock_mesh.cpp

```diff
@@ -215,32 +206,65 @@ static u16 getSmoothLightCombined(const v3s16 &p, MeshMakeData *data)
 	u16 light_day = 0;
 	u16 light_night = 0;
 
-	for (const v3s16 &dir : dirs8) {
-		MapNode n = data->m_vmanip.getNodeNoExNoEmerge(p - dir);
-
-		// if it's CONTENT_IGNORE we can't do any light calculations
-		if (n.getContent() == CONTENT_IGNORE)
-			continue;
+	static thread_local const bool edge_obstruction =
+		g_settings->getBool("smooth_lighting_edge_obstruction");
 
+	auto add_node = [&] (int i) -> const ContentFeatures& {
```

Comment:
this syntax is advanced magic

Reply:
thanks =)

Reply:
wow, i didn't know this syntax, can you explain me ? it sounds like a lambda

Reply:
`[&]` means the variable is captured by reference. You already know the rest, I think.

Reply:
@kilbith lol :) the problem is no the lambda header but the -> const ContentFeature & { }
i never used a such syntax

Reply:
@nerzhul It's the return type of the lambda written explicity, which is not necessary here, but I added it because it seemed clearer this way.